packages/flutter_tools/lib/src/daemon.dart

```diff
@@ -219,6 +219,10 @@ class DaemonStreams {
       if (binary != null) {
         _outputSink.add(binary);
       }
+    } on StateError catch (error) {
+      _logger.printError('Failed to write daemon command response: $error');
```

Comment:
@zanderso Actually, if we're running in the daemon context and STDOUT is closed, should we just tool exit since the IDE plugin isn't listening anyway?

Reply:
It may happen already as a consequence of closing `_outputSink`. But otherwise, probably yeah.